js/latoken.js

```diff
@@ -0,0 +1,715 @@
+            'cliOrdId': cliOrdId,
+            'side': side,
+            'price': price,
+            'amount': amount,
+            'orderType': orderType,
+            'timeAlive': timeAlive,
+            
+        };
+        let result = await this.privatePostOrderNew(this.extend(order, params));
+        return this.parseNewOrder(result);
+    }
+
+    async cancelOrder (id, params = {}) {
+        await this.loadMarkets();
+        if (id === undefined)
+            throw new ArgumentsRequired (this.id + ' cancelOrder requires a id argument');
+        await this.loadMarkets ();
+        let response = await this.privatePostOrderCancel (this.extend ({ 
+            'orderId': id 
+        }, params));
+        return this.parseOrder(response)
+    }
+    
+    parseAllOrders(orders) {
+        let pairId = orders['pairId'];
+        let symbol = orders['symbol']
+        let canceledOrders = [];
+        for (let i = 0; i < orders.cancelledOrders.length; i++) {
+            canceledOrders.push(
+                orders.cancelledOrders[i]
+            )
+        }
+        let result = {
+            'pairId': pairId,
+            'symbol': symbol,
+            'cancelledOrders': canceledOrders,
+        }
+        return result;
+    }
+
+    async cancelAllOrders(symbol, params = {}) {
+        await this.loadMarkets();
+        let response =  await this.privatePostOrderCancelAll(this.extend({
+            'symbol': symbol
+        }, params));
+        return this.parseAllOrders(response);
+    }
+
+    sign (path, api = 'public', method = 'GET', params = undefined, headers = undefined, body = undefined) {
+        let signature;
+        let query1;
+        let url = this.urls['api']['v1'];
+        url += '/' + path;
+        if (api === 'public') {
+            headers = {
+                'Content-type': 'application/json',
+                'x-lat-timestamp': this.nonce(),
+                'x-lat-timeframe': this.options['timeframe'],
+            };
+            if ((path === 'exchangeInfo/pairs' || 'exchangeInfo/currencies') && (typeof(params) === 'string')) {
+                url += '/' + params;                
+            }
+            url += '?' + this.urlencode (params);
+        } else  if (api === 'private') {
+            this.checkRequiredCredentials();
+            if (path === 'account/balances' && (typeof(params) === 'string')) {
+                url += '/' + params;
+                let param = {
+                    'timestamp': Date.now(),
+                }
+                query1 = `?${Object.entries(param).map(([key, val]) => `${key}=${val}`).join('&')}`
+                let dataToSign = '/api/v1/' + path + '/' + params; 
+                signature = crypto.createHmac('sha256', this.secret).update(dataToSign+query1).digest('hex');
+            } else {
+                params['timestamp'] = this.nonce();
+                query1 = `?${Object.entries(params).map(([key, val]) => `${key}=${val}`).join('&')}`
```

Comment:
`Object.entries` is not allowed. Refer to `this.urlencode` and `this.implode` instead.

`crypto` is also not allowed (because it does not work in browser). Take a look at `this.hash` instead.

Reply:
done

Reply:
done